src/datalakebundle/table/schema/tests/SchemaChecker_test.py

```diff
@@ -0,0 +1,393 @@
+                ],
+            ),
+        ),
+    ],
+)
+
+df_schema4 = t.StructType(
+    [
+        t.StructField("FIELD1", t.IntegerType()),
+    ]
+)
+
+expected_schema6 = t.StructType(
+    [
+        t.StructField(
+            "FIELD6",
+            t.StructField(
+                "STRUCT1",
+                t.StructType(
+                    [
+                        t.StructField("NESTED_FIELD0", t.StringType()),
+                    ],
+                ),
+            ),
+        ),
+    ]
+)
+
+df_schema_array = t.StructType(
+    [
+        t.StructField("FIELD1", t.IntegerType()),
+        t.StructField("FIELD2", t.DoubleType()),
+        t.StructField(
+            "ARRAY1",
+            t.ArrayType(t.StringType()),
+        ),
+    ],
+)
+
+expected_schema_array = t.StructType(
+    [
+        t.StructField("FIELD1", t.IntegerType()),
+        t.StructField("FIELD2", t.DoubleType()),
+        t.StructField(
+            "ARRAY1",
+            t.ArrayType(t.ArrayType(t.StringType())),
+        ),
+    ],
+)
+
+df_schema_array2 = t.StructType(
+    [
+        t.StructField("FIELD1", t.IntegerType()),
+        t.StructField("FIELD2", t.DoubleType()),
+        t.StructField(
+            "ARRAY1",
+            t.ArrayType(
+                t.ArrayType(
+                    t.StructType(
+                        [
+                            t.StructField("FIELD4", t.IntegerType()),
+                            t.StructField("FIELD5", t.IntegerType()),
+                        ]
+                    )
+                )
+            ),
+        ),
+    ],
+)
+
+expected_schema_array2 = t.StructType(
+    [
+        t.StructField("FIELD1", t.IntegerType()),
+        t.StructField("FIELD2", t.DoubleType()),
+        t.StructField(
+            "ARRAY1",
+            t.ArrayType(t.ArrayType(t.StringType())),
+        ),
+    ],
+)
+
+df_schema_array3 = t.StructType(
+    [
+        t.StructField("FIELD1", t.IntegerType()),
+        t.StructField("FIELD2", t.DoubleType()),
+        t.StructField(
+            "ARRAY1",
+            t.ArrayType(
+                t.StructType(
+                    [
+                        t.StructField("FIELD4", t.IntegerType()),
+                        t.StructField("FIELD5", t.IntegerType()),
+                    ]
+                )
+            ),
+        ),
+    ],
+)
+
+df_schema_array_reordered_changed_case = t.StructType(
+    [
+        t.StructField(
+            "arRay1",
+            t.ArrayType(
+                t.ArrayType(
+                    t.StructType(
+                        [
+                            t.StructField("FIELD5", t.IntegerType()),
+                            t.StructField("field4", t.IntegerType()),
+                        ]
+                    )
+                )
+            ),
+        ),
+        t.StructField("FIELD1", t.IntegerType()),
+        t.StructField("FIELD2", t.DoubleType()),
+    ],
+)
+
+
+class MyTestCase(unittest.TestCase):
```

Comment:
class MyTestCase, filename: SchemaChecker_test.py - why? Why not just SchemaCheckerTest(.py) ?

Reply:
fixed